src/ol/webgl/TileTexture.js

```diff
@@ -80,20 +80,32 @@ function uploadDataTexture(
   }
 
   let format;
+  let internalFormat;
   switch (bandCount) {
     case 1: {
       format = gl.LUMINANCE;
+      textureType = gl.UNSIGNED_BYTE;
+      data = new Uint8Array(Array.prototype.slice.call(data));
```

Comment:
The failing test uses single band Float32 data but you are you are converting that to UInt8.  Since single band Float32 data is valid and can have values beyond the range of UInt8 that seems wrong.  Should the internal formats for 1 and 2 bands be R32F and RG32F for WebGL2 if the data is Float32?